Change jailer's 'unshare' + 'exec' to 'clone' (#2652)
firecracker-microvm/firecracker: docs/jailer.md

  jailer will use this to join the associated network namespace.
- When present, the `--daemonize` flag causes the jailer to cal `setsid()` and
  redirect all three standard I/O file descriptors to `/dev/null`.

> - When present, the `--new-pid-ns` flag causes the jailer to spawn the provided

Reviewer: There is also a reference to `new-pid-ns` here: https://github.com/firecracker-microvm/firecracker/blob/a39fabbe1ab39af41a154142858ce9e981414fa5/docs/jailer.md#jailer-operation. Please update that too.

  binary into a new PID namespace. 

>   It does it using the clone libc function with the CLONE_NEWPID flag.

Reviewer 2: ```suggestion
  It makes use of the libc `clone()` function with the CLONE_NEWPID flag.
```
